deployments/deploy-tests.ps1

```diff
@@ -1,99 +1,122 @@
-    #$kv = New-KeyVaultDefinition
+    $la = New-PfLogAnalyticsBuild
+
+    Deploy-PfDeploymentContext
+
+     ##############################################################################################################
+    #
+    #               Switch Az Contexts in the middle of deployment context
+    #
+    #############################################################################################################
+
+    #Shows you can build a keyvault for the admin team, then switch context to the "DM" team and build a LA
+
+    New-PfDeploymentContext
+
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D" -Region "CentralUs"
 
     #name, az context built from deployment context params above
-    #$la = New-LogAnalyticsDefinition
+    $kv = New-PfKeyVaultBuild
+
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "DM" -EnvironmentLetter "D" -Region "CentralUs"
 
-    #$deployContext.ResourceDefinitions.AddRange(@($kv,$la))
-    #$deployContext.Deploy()
+    #name, az context built from deployment context params above
+    $la = New-PfLogAnalyticsBuild
 
+    Deploy-PfDeploymentContext
 
     ##############################################################################################################
     #
-    #               Deploy multiple resources with context and deployment context
+    #               Override the "enterprise standards"
     #
     #############################################################################################################
 
-    #Still need to write the deploy method, but the thought here is that you get the sub and rg from the
-    #current context so all you need to do is define the name and other options
-
-    $deployContext = New-PfDeploymentContext
+    #Shows how you can override the enterprise standards for any option such as name or resource group
 
-    Set-AzCloudContext -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D"
+    New-PfDeploymentContext
 
-    $kvOptions = New-KeyVaultOptions
-    $kvOptions.Name="KV-MYAEA-KKZH-ADM-C1-D01" 
-    $kv = New-KeyVaultDefinition -Options $kvOptions
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D" -Region "CentralUs"
 
-    $laOptions = New-LogAnalyticsOptions
-    $laOptions.Name="LA-MYAEA-KKZH-ADM-C1-D01" 
-    $la = New-LogAnalyticsDefinition -Options $laOptions
+    #name, az context built from deployment context params above
+    $kv = New-PfKeyVaultBuild
+    $kv.Options.Name = "testing"
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "DM" -EnvironmentLetter "D" -Region "CentralUs"
 
-    $deployContext.ResourceDefinitions.AddRange(@($kv,$la))
-    #$deployContext.Deploy()
+    #name, az context built from deployment context params above
+    $la = New-PfLogAnalyticsBuild
```

Comment:
$la = New-PfLogAnalyticsBuild
    $la.Options.ResourceGroupName="NewRg"

I think this syntax good, but wonder if someone might want to set as like key/values in future.  i think that is very much an optimization though as take time to build for each resource (this good enough)

Reply:
So you could still build the key value pairs if you wanted to.  do something like 
[PLogAnalyticsOptions]$newOptions=@{
ResourceGroupName="NewRg"
}
$la = New-PfLogAnalyticsBuild
$la.Options=$newOptions

But then we would have to right a method though that would keep defaults and only change the options supplied though